### PR TITLE
Null safety (and a few other things)

### LINKED ISSUE #72: Null-safety
Release a null-safe version of this package.

### PR DESCRIPTION
**Pull request by evanweible-wf**
- Null safety
- Removes the public `package:dependency_validator/dependency_validator.dart` entrypoint since this package is really only supposed to provide the executable and not any Dart APIs.
- Addresses some things that the [pub site calls out for this package](https://pub.dev/packages/dependency_validator/score):
  - added a small example document
  - formatted source files with default line length of 80
- lastly, switched from Travis CI to GitHub workflow.

Recommend reviewing this PR one commit at a time, as they are broken up into logical chunks.

Closes #72.

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: pubspec.yaml
Comment (travissanderson-wf) [inline pubspec.yaml:10]: what's the story with the override?

Comment (evanweible-wf) [reply pubspec.yaml:10]: Just waiting for the NNBD release of it - it's been merged, but not published:
https://github.com/dart-lang/build/blob/master/build_config/CHANGELOG.md#100-dev

I was keeping an eye on the repo and they've still go a couple of open NNBD PRs so I figured it would get published once those merge.

**Review by travissanderson-wf — APPROVED** [review]

**Comment by zmeggyesi** [comment]
@evanweible-wf Is there anything holding up this PR? Do you need anything to move this along, with the `w_transport` upgrade?

**Comment by evanweible-wf** [comment]
Yes, this PR is dependent on the `build_config` null-safe release was has not yet been released. I asked about it [here](https://github.com/dart-lang/build/issues/3073). Since it is taking a bit longer, I will probably just move forward with our other packages that use this package without waiting, which should be fine since we are only using the executable.

**Review by corwinsheahan-wf — APPROVED** [review]

**Comment by evanweible-wf** [comment]
QA +1
- [x] CI passes

@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**
